openslides/motions/models.py

```diff
@@ -771,6 +771,25 @@ class MotionChangeRecommendation(RESTModelMixin, models.Model):
     creation_time = models.DateTimeField(auto_now=True)
     """Time when the change recommendation was saved."""
 
+    def collides_with_other_recommendation(self):
+        recommendations = (MotionChangeRecommendation.objects
+                           .filter(motion_version=self.motion_version)
+                           .exclude(pk=self.pk))
+
+        for recommendation in recommendations:
+            if not ((self.line_from < recommendation.line_from and self.line_to <= recommendation.line_from) and
+                    not (self.line_from >= recommendation.line_to and self.line_to > recommendation.line_to)):
+                return True
+
+        return False
+
+    def save(self, *args, **kwargs):
+        if self.collides_with_other_recommendation():
+            raise WorkflowError('The recommendation collides with an existing one (line %s - %s).' %
```

Comment:
Please do not use `WorkflowError` because this means an error concerning motion workflows (see models Workflow and State) and should not have anything to to with change recommendations.

You can add a new exception in motions/exceptions.py and optionally catch this exception somewhere in the change recommendation viewset. Or even better use a ValidationError. This should return a 400 code in the viewset.